airflow/models.py

```diff
@@ -119,12 +119,17 @@ def clear_task_instances(tis, session, activate_dag_runs=True):
             if ti.job_id:
                 ti.state = State.SHUTDOWN
                 job_ids.append(ti.job_id)
-        # todo: this creates an issue with the webui tests
-        # elif ti.state != State.REMOVED:
-        #     ti.state = State.NONE
-        #     session.merge(ti)
         else:
-            session.delete(ti)
+            try:
+                ti_retries = dag.get_task(ti.task_id).retries
```

Comment:
Let's rename this to task_retries it's a bit clearer (otherwise it's not really clear what the difference between ti_retries and ti.max_tries is)